source/PlayerInfoPanel.cpp

```diff
@@ -766,34 +833,65 @@ void PlayerInfoPanel::DrawFleet(const Rectangle &bounds)
 			: dim
 		);
 
-		// Indent the ship name if it is a fighter or drone.
-		table.Draw(ship.CanBeCarried() ? "    " + ship.Name() : ship.Name());
-		table.Draw(ship.DisplayModelName());
-
-		const System *system = ship.GetSystem();
-		table.Draw(system ? (player.KnowsName(*system) ? system->Name() : "???") : "");
-
-		string shields = to_string(static_cast<int>(100. * max(0., ship.Shields()))) + "%";
-		table.Draw(shields);
-
-		string hull = to_string(static_cast<int>(100. * max(0., ship.Hull()))) + "%";
-		table.Draw(hull);
-
-		string fuel = to_string(static_cast<int>(
-			ship.Attributes().Get("fuel capacity") * ship.Fuel()));
-		table.Draw(fuel);
+		std::vector<std::string> row;
+		for(const auto &column : columns)
+		{
+			if(visibleColumns.find(column.name) == visibleColumns.end())
+				continue;
+
+			// Decided an if-else chain was less faff than using a switch on a std::string hash, or an enum.
```

Comment:
in a future PR I intend to replace this with a lambda in the SortableColumn instance.